src/main/java/org/springframework/metrics/instrument/prometheus/CustomPrometheusLongTaskTimer.java

```diff
@@ -0,0 +1,121 @@
+    private final String name;
+    private final List<String> tagKeys;
+    private final Collection<Child> children = new ConcurrentLinkedQueue<>();
+
+    public CustomPrometheusLongTaskTimer(String name, List<String> tagKeys, Clock clock) {
+        this.clock = clock;
+        this.name = name;
+        this.tagKeys = tagKeys;
+    }
+
+    Child child(Iterable<Tag> tags) {
+        Child child = new Child(tags);
+        children.add(child);
+        return child;
+    }
+
+    class Child implements CustomCollectorChild {
+        private final List<String> tagValues;
+        private final ConcurrentMap<Long, Long> tasks = new ConcurrentHashMap<>();
+        private final AtomicLong nextTask = new AtomicLong(0L);
+
+        Child(Iterable<Tag> tags) {
+            this.tagValues = stream(tags.spliterator(), false).map(Tag::getValue).collect(toList());
+        }
+
+        @Override
+        public Stream<MetricFamilySamples.Sample> collect() {
+            Stream.Builder<MetricFamilySamples.Sample> samples = Stream.builder();
+
+            List<String> activeTasksKeys = new LinkedList<>(tagKeys);
+            activeTasksKeys.add("statistic");
+            List<String> activeTasksValues = new LinkedList<>(tagValues);
+            activeTasksValues.add("activeTasks");
+            samples.add(new MetricFamilySamples.Sample(name, activeTasksKeys, activeTasksValues, activeTasks()));
+
+            List<String> durationKeys = new LinkedList<>(tagKeys);
+            durationKeys.add("statistic");
+            List<String> durationValues = new LinkedList<>(tagValues);
+            durationValues.add("duration");
+            samples.add(new MetricFamilySamples.Sample(name, durationKeys, durationValues, duration()));
+
+            return samples.build();
+        }
+
+        public long start() {
+            long task = nextTask.getAndIncrement();
+            tasks.put(task, clock.monotonicTime());
+            return task;
+        }
+
+        public long stop(long task) {
+            Long startTime = tasks.get(task);
+            if (startTime != null) {
+                tasks.remove(task);
+                return clock.monotonicTime() - startTime;
+            } else {
+                return -1L;
+            }
+        }
+
+        public long duration(long task) {
+            Long startTime = tasks.get(task);
+            return (startTime != null) ? (clock.monotonicTime() - startTime) : -1L;
+        }
+
+        public long duration() {
+            long now = clock.monotonicTime();
+            long sum = 0L;
+            for (long startTime : tasks.values()) {
+                sum += now - startTime;
+            }
+            return sum;
+        }
+
+        public int activeTasks() {
+            return tasks.size();
+        }
+    }
+
+    @Override
+    public List<MetricFamilySamples> collect() {
+        return Collections.singletonList(new MetricFamilySamples(name, Type.UNTYPED, " ", children.stream()
```

Comment:
I'm getting the feeling that types are largely for documentation purposes.
https://prometheus.io/docs/instrumenting/writing_exporters/#types

I think `UNTYPED` is a fine default.

At a meetup I recall them mentioning how 'someday they would like to do something with `#HELP`, I suspect `#TYPE` is similar. And only counters are really interesting because then the user knows they can count on the value going up and using `rate` or `increase` makes sense.